### ISSUE TITLE
There are several necessary files that do not exist in the directory, such as losses_AMA

### ISSUE THREAD

**Issue by YXPengl** [Hedawl/PINO-MBD#3]
Dear WenHaoDing
Hello, when I tried to run it, I found that some necessary files such as eval_2d and losses_AMA were missing. Could you please provide these documents?
Thank you for your reply！

**Comment by WenHaoDing** [comment]
Hi YXPengI,

I don't think you need those files to reproduce the results in the paper. eval_2d and losses_AMA were for another project. Which script are you trying to run?

**Comment by WenHaoDing** [comment]
You could send your infomation to my email account: wd123@ic.ac.uk
We could arrange a meeting to solve this problem more efficiently

**Comment by gulnazkanwal086-sud** [comment]
Dear WenHaoDing,

I am trying to run the script for VTCD (GradNorm / ODE-magnitude weighting path, i.e., Algorithm 3). When I execute the following command:
python .\Runner_VTCD_GradNorm.py --config_path configs/VTCD/VTCD.yaml --mode train
During training, the train_2d.py file imports AMA_PINO_loss from losses_AMA, but these files appear to be missing. Could you please let me know whether it’s possible to run the code without these files? If yes, what kind of modifications would be required?

Additionally, can this script be run on a CPU?

Thank you for your help.

**Comment by WenHaoDing** [comment]
> Dear WenHaoDing,
> 
> I am trying to run the script for VTCD (GradNorm / ODE-magnitude weighting path, i.e., Algorithm 3). When I execute the following command: python .\Runner_VTCD_GradNorm.py --config_path configs/VTCD/VTCD.yaml --mode train During training, the train_2d.py file imports AMA_PINO_loss from losses_AMA, but these files appear to be missing. Could you please let me know whether it’s possible to run the code without these files? If yes, what kind of modifications would be required?
> 
> Additionally, can this script be run on a CPU?
> 
> Thank you for your help.

Hi,

I don't think you need AMA loss for the VTCD problem. I guess you could just add try... except... when you run into these importing issues.

Currently my research interest lies in medical imaging and CFD. In the foreseeable future, I'm not planning to maintain this repository. May I suggest you to take a look at this:

https://github.com/NVIDIA/physicsnemo.git

This package is developed by Nvidia. I've been working with this for the past few years and I would say this is the best open-source toolkit you can find for physics-informed training. You can find a family of neural operators in there ready to be imported.

For the GPU issue, you can always run Pytorch scripts with CPU. Usually that would make things slower by 7-8 times. May I suggest try using Google Colab as it's free and you get nice GPUs with large memory. Or you could just rent GPUs these days, which I find easy and not too expensive.

Regards,
Wenhao

**Comment by gulnazkanwal086-sud** [comment]
thank you

**Comment by gulnazkanwal086-sud** [comment]
Hi, WenHaoDing ,

I’m trying to reproduce the reported results for VTCD. I noticed that the training file 10000v2.mat is quite large, and I wanted to confirm — to achieve the exact results shown in the paper/repo, do I need to train using the complete dataset, or is it possible to reproduce similar performance with a subset?

If there’s a recommended approach for handling large .mat files (for example, loading in chunks or using preprocessed splits), I’d really appreciate any guidance.

Thanks in advance for your help!

**Comment by WenHaoDing** [comment]
> Hi, WenHaoDing ,
> 
> I’m trying to reproduce the reported results for VTCD. I noticed that the training file 10000v2.mat is quite large, and I wanted to confirm — to achieve the exact results shown in the paper/repo, do I need to train using the complete dataset, or is it possible to reproduce similar performance with a subset?
> 
> If there’s a recommended approach for handling large .mat files (for example, loading in chunks or using preprocessed splits), I’d really appreciate any guidance.
> 
> Thanks in advance for your help!

Hi,

For multi-body dynamics problem, the low-frequency behaviors are more difficult to predict, mainly due to the properties of DFT (less resolution in the low-frequency part when x is log-scale). If you reduce the dataset by half, you wouldn't have much difference in the prediction performance of low-mass bodies such as the wheels, but the car-body displacement will be a little bit blury. 

Regards,
Wenhao
